src/screens/home.tsx

```diff
@@ -1,17 +1,162 @@
-    <Button mode="contained" onPress={() => navigation.navigate('Details')}>
-      Go to details
-    </Button>
-  </View>
-)
+export type Departure = {
+  destination: string
+  waitTime: number
+  platformNumber: number
+  isSingle: boolean
+}
+export type Station = {
+  value: Board[]
+}
+
+type Board = {
+  Wait0: string
+  Dest0: string
+  Carriages0: string
+  Wait1: string
+  Dest1: string
+  Carriages1: string
+  Wait2: string
+  Dest2: string
+  Carriages2: string
+  Wait3: string
+  Dest3: string
+  Carriages3: string
+  AtcoCode: string
+}
+const HomeScreen = ({ navigation }: HomeScreenProps) => {
+  const [details, setDetails] = useState<Departure[]>([])
+  const [dropDownVisible, setDropDownVisible] = useState(true)
+  const [station, setStation] = useState('')
+  const [pageContent, setPageContent] = useState(<></>)
+  const [refreshCount, setRefreshCount] = useState(0)
+  const [count, setCount] = useState(0)
+
+  useEffect(() => {
+    setPageContent(showDropDown())
+  }, [details, dropDownVisible, refreshCount])
+
+  useEffect(() => {
+    const timer = setTimeout(
+      () => !dropDownVisible && setCount(count + 1),
+      1000
+    )
+    const fetchData = async () => {
+      const filter: string =
+        '?$filter=StationLocation eq \'' + station.replace('\'', '\'\'') + "'"
+      const results = await getStopData(filter)
+      handleResults(results)
+    }
+    fetchData().then((r) => {})
+    return () => clearTimeout(timer)
+  }, [count, dropDownVisible])
+  function handleResults(stationInfo: Station) {
+    //console.log(stationInfo)
+    const boardsInfo: Board[] = stationInfo['value']
+    const departureTimes = addDepartureTimes(boardsInfo)
+
+    departureTimes.sort((a, b) => {
+      return a.waitTime - b.waitTime
+    })
+
+    setDetails(departureTimes)
+  }
+
+  function addDepartureTimes(boardsInfo: Board[]) {
+    const departureTimes: Departure[] = []
+    const handledPlatforms: Set<string> = new Set()
+    for (const boardInfo of boardsInfo) {
+      //console.log(boardInfo)
+      if (handledPlatforms.has(boardInfo['AtcoCode'])) {
+        continue
+      }
+      handledPlatforms.add(boardInfo['AtcoCode'])
+      for (let i = 0; i <= 3; i++) {
+        const waitKey = 'Wait' + i
+        const destKey = 'Dest' + i
+        const carriagesKey = 'Carriages' + i
+        const platformChar =
+          boardInfo['AtcoCode'][boardInfo['AtcoCode'].length - 1]
+        const waitTimeString: string = boardInfo[waitKey]
+        if (waitTimeString.length === 0) {
+          break
+        }
+        departureTimes.push({
+          destination: boardInfo[destKey],
+          waitTime: +waitTimeString,
+          platformNumber:
+            platformChar <= '9' && platformChar >= '1' ? platformChar - '0' : 1,
+          isSingle: boardInfo[carriagesKey] === 'Single',
+        })
+      }
+    }
+    return departureTimes
+  }
+
+  function showDropDown(): JSX.Element {
```

Comment:
so instead of putting the component in the state and returning the state, just return the components returned by `showDropDown`.